src/Features/Core/Portable/EditAndContinue/EditAndContinueCapabilities.cs

```diff
@@ -50,6 +50,21 @@ internal enum EditAndContinueCapabilities
         /// Whether the runtime supports updating the Param table, and hence related edits (eg parameter renames)
         /// </summary>
         UpdateParameters = 1 << 6,
+
+        /// <summary>
+        /// Adding a static or instance method, property or event to an existing type (without backing fields), such that the method and/or the type are generic.
+        /// </summary>
+        GenericAddMethodToExistingType = 1 << 7,
+
+        /// <summary>
+        /// Updating an existing static or instance method, property or event (without backing fields) that is generic and/or contained in a generic type. 
+        /// </summary>
+        GenericUpdateMethod = 1 << 8,
+
+        /// <summary>
+        /// Adding a static or instance field to an existing generic type.
+        /// </summary>
+        GenericAddFieldToExistingType = 1 << 9,
```

Comment:
If these haven't been committed elsewhere yet, I'd consider renaming them to move the word "Generic". As nice as it is to group them together, I think it would be more readable to describe the type/method as being generic, rather than right now where it makes it sound like the update itself is generic in nature.

Reply:
Yeah, I thought about that. The problem is that the `UpdateGenericMethod` is not about just updating method that has type parameters, but also about updating a regular method in a generic type. Such method is not usually called "generic".
"Generic" in all these cases refers to a "generic context".